kv.go

```diff
@@ -95,21 +95,42 @@ type KeyWatcher interface {
 	Stop() error
 }
 
-type CancelOpt interface {
-	configureCancel(opts *cancelOpts) error
+// KeyValueOption configures the Key-Value store instance
+type KeyValueOption func(*kvs) error
+
+// BucketSubjectPrefix sets a custom prefix to use for write operations like Put() to the bucket.
+// Usually keys are accessed as $KV.BUCKET.KEY, this would replace the $KV.BUCKET. part of
+// that subject.
+//
+// When a APIPrefix() was used to create the JetStream context the KV Stream API and Subjects
```

Comment:
Still a bit concerned about monkeying with data subjects outside of the server itself through imports.

Reply:
I do believe there should be an easy way to match service import mappings into the client.

Reply:
This is essentially a stop gap and to support people with weird stuff.  Matthias wants to add server-side helpers that would mean things would have fewer moving parts and more strict convention.

Right now though, KV simply can't be used cross account at all - so until above is finalized we need something. 

Can you expand on what your concerns are and what you mean with matching imports? We cannot overlap subjects and someone might want to import 2 x FOO buckets from different accounts.

Reply:
To date we have not monkeyed with the data subjects, only control via APIPrefix etc. 

This is a departure from that, so I just want us to be very sure about this path.

Reply:
Agree, personally do not really like the proposed structure in ADR-19, would rather just keep to arbitrary prefix for the moment while we work on that.  However it gets super tedious to require both ApiPrefix() and BucketSubjectPrefix() so figured I might default to the adr behavior.

I am do not like at all what the adr propose where it has `fromA.STREAM.INFO.FOO` and `fromA.$KV.B.K` the fact that it mounts these 2 subjects speaces to overlap each other is quite funky.

So I had it without the ApiPrefix() handling at first, but would also rather remove it while we work out ADR-19

Reply:
Removed the ApiPrefix() handling, we should remove ADR-19 while we go back to the drawing board on that.